### PR TITLE
#1657 Add python version detection function and call

### PR DESCRIPTION
**Pull request by kaliGo-Li**
# What changes were proposed in this pull request?
- [ ] Finshed changes describe
make-distribution.sh add python version detection function and call.
# How was this patch tested?
- [ ] Testing done
截屏2022-02-11 下午3 43 21
截屏2022-02-11 下午3 43 56

# Are there and DOC need to update?
- [ ] Doc is finished
Do not need to update DOC。
# Spark Core Compatibility

### CODE REVIEW

**Comment by hellozepp** [comment]
@kaliGo-Li Please make sure to revert our dependencies to the `spark3` version.
You can specify MLSQL_SPARK_VERSION version as 3.0 when running `make-distribution.sh`.

File: external/mlsql-autosuggest/pom.xml [hunk 1]
```diff
@@ -3,27 +3,27 @@
...
+         <scalatest.version>3.0.3</scalatest.version> 
         <!-- spark 2.4 end -->
 
         <!-- spark 3.0 start -->
-        <scala.version>2.12.10</scala.version> 
-        <scala.binary.version>2.12</scala.binary.version> 
-        <spark.version>3.1.1</spark.version> 
-        <spark.bigversion>3.0</spark.bigversion> 
-        <scalatest.version>3.0.3</scalatest.version> 
+<!--         <scala.version>2.12.10</scala.version>  -->
```
Comment (hellozepp) [inline external/mlsql-autosuggest/pom.xml]: Do not change the maven dependency version.

**Review by hellozepp — CHANGES_REQUESTED** [review]

**Comment by hellozepp** [comment]
@kaliGo-Li Can you use rebase to squeeze your multiple commits?

File: dev/make-distribution.sh [hunk 2]
```diff
@@ -35,18 +35,14 @@ checkPython()
     V2=X
     V3=X
 
-    echo need python version is : $V1.$V2.$V3
     U_V1=`python -V 2>&1|awk '{print $2}'|awk -F '.' '{print $1}'`
```
Comment (hellozepp) [inline dev/make-distribution.sh]: Please use parameter names that are easy to understand

File: dev/make-distribution.sh [hunk 3]
```diff
@@ -28,7 +28,29 @@ ENABLE_HIVE_THRIFT_SERVER        Hive ThriftServer                default true
...
+{
+  
+    V1=3
+    V2=X
+    V3=X
 
+    echo need python version is : $V1.$V2.$V3
+    U_V1=`python -V 2>&1|awk '{print $2}'|awk -F '.' '{print $1}'`
+    U_V2=`python -V 2>&1|awk '{print $2}'|awk -F '.' '{print $2}'`
+    U_V3=`python -V 2>&1|awk '{print $2}'|awk -F '.' '{print $3}'`
```
Comment (hellozepp) [inline dev/make-distribution.sh]: Can you delete the two useless parameters U_V2, U_V2?

**Review by hellozepp — CHANGES_REQUESTED** [review]

File: dev/make-distribution.sh [hunk 4]
```diff
@@ -28,6 +28,16 @@ ENABLE_HIVE_THRIFT_SERVER        Hive ThriftServer                default true
...
+checkPython(){
+    python_version=3
+
+    user_py_version=`python -V 2>&1|awk '{print $2}'`
+    if [ "${user_py_version%%.*}" -lt $python_version ]; then
+        echo Your python version is : $user_py_version
+        echo 'Current python version is not supported! Requires python version is 3.x.x.'
+        exit 1
+    fi
+}
```
Comment (hellozepp) [inline dev/make-distribution.sh:41]: Should we call this function `checkPython` ?

**Comment by hellozepp** [comment]
@kaliGo-Li good job!  LGTM
@allwefantasy PTAL

**Comment by hellozepp** [comment]
image

**Review by hellozepp — APPROVED** [review]

**Comment by allwefantasy** [comment]
LGTM

**Review by allwefantasy — APPROVED** [review]

**Pull request merged**